### ISSUE TITLE
bug: `parseUnits` strips leading zero's of the factional causing wrong results

### ISSUE THREAD

**Issue by magiclars-off** [wagmi-dev/viem#634]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package Version

0.3.43

### Current Behavior

`parseUnits("630.0056882018259", 6)` results in `6305688n`

### Expected Behavior

`parseUnits("630.0056882018259", 6)` results in `630005688n`

### Steps To Reproduce

see above

**Comment by jxom** [comment]
Thanks! Fixed in `0.3.49`.

**Comment by xenoliss** [comment]
Hey, seems like the error was reintroduced ? I'm on `"viem": "^1.0.2"` and running:
```js
const decimal = 1.0536059576998882;
const bn = Number(parseUnits(`${decimal}`, 7));
console.log(bn);
console.log(bn / 1e7);
```
outputs:
```
1.0536059576998882
1536060
0.153606
```
